### DISCUSSION TITLE
Issues Using Free Font With Sprite

### DISCUSSION THREAD

**Discussion by audreylim** [Bodmer/TFT_eSPI#1729] [New processor/display support or capability request] [answered]
I'm trying to display changing numbers without flickering. Following this example and using NotoSans font worked:

https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Smooth%20Fonts/FLASH_Array/Font_Demo_4_Array/Font_Demo_4_Array.ino

```
  spr.loadFont(AA_FONT_LARGE);
  spr.setTextColor(TFT_WHITE, TFT_BLACK);
  tft.setCursor(0, 95); 
  spr.printToSprite("123");
  spr.unloadFont();
```

However when trying to use free font, the following didn't work:

```
 #define FF_LARGE &FreeSans18pt7b

  tft.setFreeFont(FF_LARGE);
  spr.setTextColor(TFT_WHITE, TFT_BLACK);
  spr.drawString("123", 0, 95, GFXFF);
```
and
```
  #define FF_LARGE &FreeSans18pt7b

  tft.setFreeFont(FF_LARGE);
  tft.setCursor(0, 95); 
  spr.printToSprite("123");
```

Is it possible to use free font with sprites? What am I missing?

**Comment by Bodmer** [comment]
I suspect if you use:
```
spr.setFreeFont(FF_LARGE);
```

It will work.

Since the text is printed in the sprite first, it is the sprite that needs to have the font loaded.

I know this can get confusing...

**Comment by audreylim** [comment]
Oh, yes I also tried `spr.setFreeFont(FF_LARGE)` and it didn't work...

**Comment by Bodmer** [comment, marked as answer]
Ah, yes. I have just checked and printToSprite only works for smooth fonts!

You can print a free font in a sprite that has already been created.

Example[ here](https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Sprite/Sprite_scroll_16bit/Sprite_scroll_16bit.ino#L121-L149).

**Comment by audreylim** [comment]
That works, thank you! It'll be great if printToSprite supports free fonts too.

**Comment by Bodmer** [comment]
I tend to add features to support my own projects and I got lazy when I added printToSprite. I will add to my (long) to do list!

**Reply by audreylim** [reply]
Ah yes, no worries and thank you for sharing your library! :)
